test/rulesets/LinearAlgebra/dense.jl

```diff
@@ -119,6 +119,12 @@
                 test_rrule(f, B)
             end
         end
+        @testset "$f(unitary matrix)" begin
+            B = generate_well_conditioned_matrix(ComplexF64, 4)
+            U = exp(1im * (B + B'))
```

Comment:
Does `B` need to be well-conditioned? Because if `Z` in `U=exp(Z)` is skew-Hermitian, then `U` is well-conditioned.

```suggestion
            B = randn(ComplexF64, 4, 4)
            U = exp(B - B')
```

Reply:
`B` does not have to be well-conditioned... in fact, `B` can be completely random. Any unitary matrix is well-conditioned (same as your `U=exp(B-B')`, I think). My actual use case that found this bug involves unitary matrices, so that's why I went with a unitary test matrix.

The only reason I was using the `B` as a "seed" is that it delegates to `generate_well_conditioned_matrix(Random.GLOBAL_RNG, T, N)` in `ChainRulesTestUtils`, and I wasn't quite sure what the `GLOBAL_RNG` is about. I thought maybe the test-suite initializes the RNG so that each test run is reproducible (uses the same random numbers each time). Putting `B = randn(ComplexF64, 4, 4)` might break that behavior, but then I just realized that plenty of other places in the test also use `randn` directly.

So yeah, if `B = randn(ComplexF64, 4, 4)` is fine, then I can use either `U = exp(1im * (B + B'))` or (presumably) `U = exp(B - B')`

Reply:
`Random.GLOBAL_RNG` is what is used by the `rand`/`randn` functions in `Random` if no `rng` is provided, so using `randn` here does the same thing. And yes, we also use `randn` with no specified `rng` through our test suite.

> So yeah, if `B = randn(ComplexF64, 4, 4)` is fine, then I can use either `U = exp(1im * (B + B'))` or (presumably) `U = exp(B - B')`

It's not very important, but the latter requires one less allocation, so I'd prefer it. The two expressions both target the same distribution on the unitary matrices.

Reply:
Ok, so I switched `B` to a random matrix, but left `U` to be unitary. Let me know if that's ok.